Cargo.toml

```diff
@@ -8,13 +8,14 @@ description = "A parallel and incremental verifier for Metamath databases"
 repository = "https://github.com/david-a-wheeler/metamath-knife"
 keywords = ["theorem", "proving", "verifier", "proof", "assistant"]
 categories = ["command-line-utilities", "development-tools", "mathematics"]
-edition = "2018"
+edition = "2021"
```

Comment:
That's right. The most notable changes in 2021 are: `[a, b, c].into_iter()` actually yields the elements by value as you would expect rather than silently getting transformed into `[a, b, c].iter()`; and closures now capture only the paths that are needed for the closure, for example `|| x.foo.clone()` captures `&x.foo` instead of `&x` like it used to. (The closure capture difference is why I enabled it in this PR, although the PR compiles without it so maybe it was in code that has since been refactored away.) Also `TryFrom` and `TryInto` are in the prelude now which is nice.